org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/services/LanguageClient.java

```diff
@@ -85,4 +87,14 @@
 	 */
 	@JsonNotification("window/logMessage")
 	void logMessage(MessageParams message);
+
+	/**
+	 * The workspace/workspaceFolders request is sent from the server to the client
+	 * to fetch the current open list of workspace folders.
+	 * @return null in the response if only a single file is open in the tool,
+	 *         an empty array if a workspace is open but no folders are configured,
+	 *         the workspace folders otherwise.
+	 */
+	@JsonRequest("workspace/workspaceFolders")
+	CompletableFuture<List<WorkspaceFolder>> workspaceFolders();
```

Comment:
Adding a method here breaks existing implementations. Since the protocol extension is still in a "proposed" state, we should avoid breaking changes. Either move the method to a separate interface or add a default implementation here. I prefer the second option.